conda_index/plugin.py

```diff
@@ -0,0 +1,19 @@
+"""
+Conda command plugin, provide "conda reindex" command, renamed to
+avoid clash with old "conda index" CLI.
+"""
+
+import conda.plugins
+
+def command(*args):
+     import conda_index.cli
+     return conda_index.cli.cli()
```

Comment:
```suggestion
     return conda_index.cli.cli(prog_name="conda reindex", args=args)
```

Reply:
That should suffice to tell click about the passed options

Reply:
It looks like there is an unexpected tuple with this change
```
% ~/miniconda3/envs/conda-index-foo/bin/conda reindex --help

# >>>>>>>>>>>>>>>>>>>>>> ERROR REPORT <<<<<<<<<<<<<<<<<<<<<<

    Traceback (most recent call last):
      File "/Users/dholth/miniconda3/envs/conda-index-foo/lib/python3.10/site-packages/conda/exception_handler.py", line 17, in __call__
        return func(*args, **kwargs)
      File "/Users/dholth/miniconda3/envs/conda-index-foo/lib/python3.10/site-packages/conda/cli/main.py", line 78, in main_subshell
        exit_code = do_call(args, parser)
      File "/Users/dholth/miniconda3/envs/conda-index-foo/lib/python3.10/site-packages/conda/cli/conda_argparse.py", line 142, in do_call
        result = plugin_subcommand.action(getattr(args, "_args", args))
      File "/Users/dholth/prog/conda-index/conda_index/plugin.py", line 10, in command
        return conda_index.cli.cli(prog_name="conda reindex", args=args)
      File "/Users/dholth/miniconda3/envs/conda-index-foo/lib/python3.10/site-packages/click/core.py", line 1128, in __call__
        return self.main(*args, **kwargs)
      File "/Users/dholth/miniconda3/envs/conda-index-foo/lib/python3.10/site-packages/click/core.py", line 1053, in main
        rv = self.invoke(ctx)
      File "/Users/dholth/miniconda3/envs/conda-index-foo/lib/python3.10/site-packages/click/core.py", line 1395, in invoke
        return ctx.invoke(self.callback, **ctx.params)
      File "/Users/dholth/miniconda3/envs/conda-index-foo/lib/python3.10/site-packages/click/core.py", line 754, in invoke
        return __callback(*args, **kwargs)
      File "/Users/dholth/prog/conda-index/conda_index/cli/__init__.py", line 134, in cli
        channel_index.index(
      File "/Users/dholth/prog/conda-index/conda_index/index/__init__.py", line 542, in index
        subdirs = self.detect_subdirs()
      File "/Users/dholth/prog/conda-index/conda_index/index/__init__.py", line 710, in detect_subdirs
        for subdir in os.scandir(self.channel_root)
    FileNotFoundError: [Errno 2] No such file or directory: "/Users/dholth/prog/conda-index/('--help',)"
```

Reply:
Fixed by https://github.com/conda/conda-index/pull/131/commits/78c67453a5753fe2ebc193ff091d65a95911a169